src/main/java/dev/openfeature/sdk/EventEmitter.java

```diff
@@ -0,0 +1,108 @@
+package dev.openfeature.sdk;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+
+import javax.annotation.Nullable;
+
+import dev.openfeature.sdk.internal.AutoCloseableLock;
+import dev.openfeature.sdk.internal.AutoCloseableReentrantReadWriteLock;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Event emitter construct to be used by providers.
+ */
+@Slf4j
+public class EventEmitter {
+
+    private static final ExecutorService taskExecutor = Executors.newCachedThreadPool();
```

Comment:
This thread pool is used to execute the handlers... I don't think it's huge deal, but it's possible that `newCachedThreadPool` could lead to sudden increased in resource consumption if lots of handlers are fired at the same time with some event. Having a fixed or single thread pool could reduce the changes of this happening, but we were already using a cached thread pool elsewhere.

Reply:
This should be fine as long as we do not have a lot of callbacks as we execute them concurrently.

However, one thing we should consider is to introduce a timeout for callbacks. I am introducing this in go-sdk and I think this is needed to avoid any performant impacts.